django/test/signals.py

```diff
@@ -172,3 +173,24 @@ def auth_password_validators_changed(**kwargs):
 def user_model_swapped(**kwargs):
     if kwargs['setting'] == 'AUTH_USER_MODEL':
         apps.clear_cache()
+        try:
+            from django.contrib.auth import get_user_model
+            UserModel = get_user_model()
+        except ImproperlyConfigured:
+            # Some tests set an invalid AUTH_USER_MODEL.
+            pass
+        else:
+            from django.contrib.auth import backends
```

Comment:
I find it more unfortunate that tests were relying on `get_user_model()` being called every time Django needs to do something with the user model and reflecting the current value of `AUTH_USER_MODEL`.

This is very inconsistent with the expectation that you can't change `AUTH_USER_MODEL` over the lifetime of a code base, or the migration history will blow up, let alone the lifetime of a process.

Third-party packages should benefit from this signal automatically so their tests shouldn't be affected.